src/main/java/de/cyface/collector/handler/UserCreationHandler.java

```diff
@@ -64,17 +66,18 @@ public void handle(final RoutingContext event) {
         final JsonObject body = event.getBodyAsJson();
         final String username = body.getString("username");
         final String password = body.getString("password");
+        final String role = body.getString("role");
 
-        mongoAuth.insertUser(username, password, new ArrayList<>(), new ArrayList<>(), ir -> {
-            if (ir.succeeded()) {
-                LOGGER.info("Added new user with id: " + username);
-                event.response().setStatusCode(201).end();
-            } else {
-                LOGGER.error("Unable to create user with id: " + username, ir.cause());
-                event.fail(400);
-            }
-        });
-
+        mongoAuth.insertUser(username, password,
+                role == null || role.isEmpty() ? new ArrayList<>() : Collections.singletonList(role),
```

Comment:
This allows the role parameter to be optional (or empty). Or should the Management API enforce that a "role" parameter is provided (empty is fine as not every user needs a role currently)?

Reply:
It would most likely be better to give everyone a default role like guest, if nothing else is provided. This role should be created on first run like the admin user.

Reply:
Roles are just Strings. There is currently no "role" collection or such a thing. I don't see why there is currently a need for that, so I now (new commit) just assign the default role string "guest" to new users without a role.

Reply:
I'll also assign that role to all existing users without explicit role.
Now the question is:
- should I assign an "admin" role to the admin user (both manually for existing and automatically for the newly created admin use on creation)?

I assume the "admin" role may also be assigned via the management API, as this API is not public?